src/System.Private.ServiceModel/tests/Stress/SharedPoolsOfWCFObjects/SharedPoolsOfWCFObjects.cs

```diff
@@ -336,40 +475,45 @@ public static void RecyclePooledFactories()
         private static CallStats s_CloseFactoryStats = new CallStats();
+
+        private static ITestTemplate<ChannelType> s_test;
+
         private static PoolOfAsyncThings<
             FactoryAndPoolOfItsAsyncObjects<
-                ChannelFactory<WcfService1.IService1>,
-                OpenAsyncOnceChannelWrapper<WcfService1.IService1>>
+                ChannelFactory<ChannelType>,
+                OpenAsyncOnceChannelWrapper<ChannelType>>
             > s_recyclablePooledFactoriesAndChannels;
 
         static RecyclablePooledFactoriesAndChannelsAsync_OpenOnce()
         {
+            s_test = new TestTemplate();
             s_recyclablePooledFactoriesAndChannels = StaticDisposablesHelper.AddDisposable(
                 new PoolOfAsyncThings<
-                    FactoryAndPoolOfItsAsyncObjects<ChannelFactory<WcfService1.IService1>, OpenAsyncOnceChannelWrapper<WcfService1.IService1>>>(
-                    maxSize: 3, // # of pooled FactoryAndPoolOfItsChannels
-                    createInstance: () => new FactoryAndPoolOfItsAsyncObjects<ChannelFactory<WcfService1.IService1>, OpenAsyncOnceChannelWrapper<WcfService1.IService1>>(
-                        createFactoryInstance: () =>
-                            TestHelpers.CreateChannelFactory<WcfService1.IService1>(TestHelpers.CreateEndPointAddress(), TestHelpers.CreateBinding()),
-                        destroyFactoryInstanceAsync: async (chf) =>
-                            await s_CloseFactoryStats.CallAsyncFuncAndRecordStatsAsync(TestHelpers.CloseFactoryAsync, chf),
-                        maxPooledObjects: 3, // # of pooled channels within each pooled FactoryAndPoolOfItsAsyncObjects
-                        createObject: (chf) =>
-                            s_CreateChannelStats.CallFuncAndRecordStats(func: () => new OpenAsyncOnceChannelWrapper<WcfService1.IService1>(TestHelpers.CreateChannel(chf))),
-                        destroyObjectAsync: async (chWr) =>
-                            await s_CloseChannelStats.CallAsyncFuncAndRecordStatsAsync(func: () => TestHelpers.CloseChannelAsync(chWr.Channel))
-                    ),
+                    FactoryAndPoolOfItsAsyncObjects<ChannelFactory<ChannelType>, OpenAsyncOnceChannelWrapper<ChannelType>>>(
+                        maxSize: 3, // # of pooled FactoryAndPoolOfItsChannels
+                        createInstance: () => new FactoryAndPoolOfItsAsyncObjects<ChannelFactory<ChannelType>, OpenAsyncOnceChannelWrapper<ChannelType>>(
+                            createFactoryInstance:
+                                s_test.CreateChannelFactory,
+                            destroyFactoryInstanceAsync: async (chf) =>
+                                await s_CloseFactoryStats.CallAsyncFuncAndRecordStatsAsync(s_test.CloseFactoryAsync, chf),
+                            maxPooledObjects: 3, // # of pooled channels within each pooled FactoryAndPoolOfItsAsyncObjects
```

Comment:
For future consideration and @mconnew  may have some recommendations here.  We generally use Task.GetAwaiter().GetResult() rather than raw 'await'.  Not required for merge.